### PR TITLE
Add Metrics To New Auth Endpoint

### PR DESCRIPTION
**Pull request by mloberg**
Also add additional tags for better metrics

### CODE REVIEW

File: app/controllers/v0/sessions_controller.rb [hunk 1]
```diff
@@ -213,5 +214,14 @@ def context_key
     rescue
       'unknown'
     end
+
+    def benchmark_tags(success = true)
```
Comment (jkassemi) [inline app/controllers/v0/sessions_controller.rb]: minor nit, but I'd prefer to see this parameter as either a k/v pair or `*tags` that can be added to the tags array in addition to the context, loa and multifactor tags.

File: app/controllers/v0/sessions_controller.rb [hunk 2]
```diff
@@ -99,12 +99,12 @@ def saml_callback
...
+        Benchmark::Timer.stop(TIMER_LOGIN_KEY, @saml_response.in_response_to, tags: benchmark_tags)
         StatsD.increment(STATSD_CALLBACK_KEY, tags: ['status:success', "context:#{context_key}"])
       else
         handle_login_error
         redirect_to Settings.saml.relay + '?auth=fail'
-        Benchmark::Timer.stop(TIMER_LOGIN_KEY, @saml_response.in_response_to, tags: ['status:fail'])
+        Benchmark::Timer.stop(TIMER_LOGIN_KEY, @saml_response.in_response_to, tags: benchmark_tags(false))
       end
     ensure
       StatsD.increment(STATSD_LOGIN_TOTAL_KEY)
```
Comment (jkassemi) [inline app/controllers/v0/sessions_controller.rb]: Partially out of context, but can we get these tags on this counter, too?

File: app/controllers/v0/sessions_controller.rb (hunk 1 above)
Comment (mloberg) [reply app/controllers/v0/sessions_controller.rb]: I'm not sure I'm following 100% here. Can you give more detail?

Comment (jkassemi) [reply app/controllers/v0/sessions_controller.rb]:
```
def benchmark_tags(*tags)
      tags << "context:#{context_key}"
      tags << "loa:#{@current_user.loa[:current]}" if @current_user
      tags << "multifactor:#{@current_user.multifactor}" if @current_user
      tags
end
```

and then

`benchmark_tags('status:success')`/`benchmark_tags('status:failure')`.

File: app/controllers/v0/sessions_controller.rb [hunk 3]
```diff
@@ -213,5 +214,14 @@ def context_key
     rescue
       'unknown'
     end
+
+    def benchmark_tags(success = true)
+      tags = []
+      tags << "status:#{success ? 'success' : 'failure'}"
+      tags << "context:#{context_key}"
+      tags << "loa:#{@current_user.loa[:current]}" if @current_user
+      tags << "multifactor:#{@current_user.multifactor}" if @current_user
```
Comment (patrickvinograd) [inline app/controllers/v0/sessions_controller.rb]: is a tag value of `nil` a problem, or will statsd handle that okay? I don't think `loa` or `multifactor` ought to be nil in any of these callback paths but I want to make sure if they are they don't blow up here in the statsd code.

Comment (mloberg) [reply app/controllers/v0/sessions_controller.rb]: If multifactor is nil, the tag will be `multifactor:nil`. If loa is nil, that might be an issue. Is that ever the case?

File: app/controllers/v0/sessions_controller.rb (hunk 1 above)
Comment (mloberg) [reply app/controllers/v0/sessions_controller.rb]: Ok, I like that better. Will update

File: app/controllers/v0/sessions_controller.rb (hunk 3 above)
Comment (patrickvinograd) [reply app/controllers/v0/sessions_controller.rb]: The loa dict should not be nil, loa[:current] also shouldn't but I could see it happening. But in that case the tag will just be loa:nil. I think we're okay here.

**Pull request merged**